documentation/src/main/asciidoc/user_guide/clustering.adoc

```diff
@@ -1086,6 +1095,30 @@ dcc.clustering().partitionHandling()
                     .whenSplit(PartitionHandling.ALLOW_READ_WRITES)
                     .mergePolicy(new CustomMergePolicy());
 ----
+[source,java]
+----
+public class CustomMergePolicy implements EntryMergePolicy<String, String> {
+
+   @Override
+   public CacheEntry<String, String> merge(CacheEntry<String, String> preferredEntry, List<CacheEntry<String, String>> otherEntries) {
+      // decide which entry should be used
+      
+      return the_solved_CacheEntry;
+   }
+----
+
+===== Deploy custom merge policies to a Infinispan server instance
+To utilise a custom EntryMergePolicy implementation on the server, it's necessary for the implementation class(es) to be deployed to the server. This is accomplished by utilising the java service-provider convention and packaging the class files in a jar which has a META-INF/services/org.infinispan.conflict.EntryMergePolicy file containing the fully qualified class name of the EntryMergePolicy implementation.
+[source]
+----
+# list all necessary implementations of EntryMergePolicy with the full qualified name
+org.example.CustomMergePolicy
+----
+
+In order for a Custom merge policy to be utilised on the server, compatibility mode must be enabled if your policies semantics require access to the stored Key/Value objects. This is because without compatibility mode, cache entries in the server are stored in a marshalled format and the Key/Value objects returned to your policy would be instances of WrappedByteArray.
+Without compatibility mode the marshaller need to packaged with the merge policy and the merge policy implementation depends on the marshalling configuration which can be real complicated.
```

Comment:
I think we should remove this sentence for now as it will probably confuse all but the most advanced users. Furthermore, this sentence contradicts the previous assertion that "compatibility mode must be enabled", so if we do keep it then we must also adapt the rest of the paragraph accordingly. 

From a technical perspective another added complication is that any custom user classes would also have to be packaged with the merge policy if unmarshalling were to occur in the merge policy. Personally I don't think we should be advocating this until we have a more user-friendly offering.